prometheus/promhttp/http_test.go

```diff
@@ -26,7 +26,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/klauspost/compress/zstd"
+	orignalzstd "github.com/klauspost/compress/zstd"
```

Comment:
Why don't we use the vendored package here as well? 🤔

Reply:
Great question! I mentioned in the description, but to reiterate:

1. To reduce the vendored code by at least 50% (we only need encoding). I am pulling the original dep for test purposes only to decode for validation.
2. Extra benefit is the validation against original, stable decoder so we know we did not mess something in our vendored code that makes the result encodable only for us :p